lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -0,0 +1,405 @@
+
+    Returns a dictionary of instance information
+    about the instance started/stopped
+
+    If the instance was not able to state change,
+    "changed" will be set to False.
+
+    """
+    wait = module.params.get('wait')
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait_max = time.time() + wait_timeout
+
+    changed = False
+
+    inst = None
+    try:
+        inst = find_instance_info(client, instance_name)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] != 'NotFoundException':
+            module.fail_json(msg='Error finding instance {0}, error: {1}'.format(instance_name, e))
+
+    # Wait for instance to exit transition state before state change
+    if wait:
+        while wait_max > time.time() and inst is not None and inst['state']['name'] in ('pending', 'stopping'):
+            try:
+                time.sleep(5)
+                inst = find_instance_info(client, instance_name)
+            except botocore.exceptions.ClientError as e:
+                time.sleep(1)
+
+    # Try state change
+    if inst is not None and inst['state']['name'] != state:
+        try:
+            if state == 'running':
+                client.start_instance(instanceName=instance_name)
+            else:
+                client.stop_instance(instanceName=instance_name)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg='Unable to change state for instance {0}, error: {1}'.format(instance_name, e))
+        changed = True
+        # Grab current instance info
+        inst = find_instance_info(client, instance_name)
+
+    return (changed, inst)
+
+def core(module):
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    client = None
+    try:
+        client = boto3_conn(module, conn_type='client', resource='lightsail',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg=str(e))
+
+    changed = False
+    state = module.params['state']
+    name = module.params['name']
+
+    if state == 'absent':
+        (changed, instance_dict) = delete_instance(module, client, name)
+    elif state in ('running', 'stopped'):
+        (changed, instance_dict) = startstop_instance(module, client, name, state)
+    elif state == 'restarted':
+        (changed, instance_dict) = restart_instance(module, client, name)
+    elif state == 'present':
+        (changed, instance_dict) = create_instance(module, client, name)
+
+    module.exit_json(changed=changed, instance=instance_dict)
+
+def find_instance_info(client, instance_name):
+    inst = None
+    try:
+        inst = client.get_instance(instanceName=instance_name)
+    except botocore.exceptions.ClientError as e:
+        raise
```

Comment:
It appears that the exceptions are being handled differently depending on where this function is called, so I added a docstring.